Includegraphics Preview crashes VS Code with some files

## Preliminary questions [Required]

### Disable all the other extensions except for LaTeX Workshop, and check that you still see this issue. [Required]

You still see this issue?: **Yes**

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: **Yes**

## Describe the bug [Required]

Includegraphics Preview crashes VS Code with this PDF file: [point-flow.pdf](https://github.com/James-Yu/LaTeX-Workshop/files/5122368/point-flow.pdf)

Also, setting `"latex-workshop.intellisense.includegraphics.preview.enabled": false` in `settings.json` doesn't disable the preview.

### To Reproduce

Hover over `\includegraphics` in this minimal reproduction with the above PDF file in the same directory. This causes to Code Helper (Renderer) to use 100 % CPU and the VS Code window to freeze up.

```tex
\documentclass{article}

\usepackage{graphicx}

\begin{document}
\begin{figure}[!htb]
  \centering
  \includegraphics[width=\linewidth]{point-flow.pdf}
  \caption{MAF}
\end{figure}
\end{document}
```

### Expected behavior

A clear and concise description of what you expected to happen.




## Logs [Required]

### LaTeX Workshop Output [Required]

<details>

<summary>Extension Logs</summary>

```
[10:30:46] Initializing LaTeX Workshop.
[10:30:46] Creating PDF file watcher.
[10:30:46] Creating LaTeX Workshop http and websocket server.
[10:30:46] LaTeX Workshop initialized.
[10:30:46] Found root file from active editor: /Users/janosh/Desktop/test.tex
[10:30:46] Root file changed: from undefined to /Users/janosh/Desktop/test.tex
[10:30:46] Start to find all dependencies.
[10:30:46] Instantiating a new file watcher for /Users/janosh/Desktop/test.tex
[10:30:46] Creating Bib file watcher.
[10:30:46] Parsing a file and its subfiles: /Users/janosh/Desktop/test.tex
[10:30:46] Parse fls file.
[10:30:46] Cannot find fls file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:30:46] Server created on 127.0.0.1:54266
[10:30:46] Added to file watcher: /Users/janosh/Desktop/test.tex
[10:30:47] LaTeX Workshop version: 8.11.1
[10:30:47] Snippet data loaded.
[10:34:23] Found root file from active editor: /Users/janosh/Desktop/test.tex
[10:34:23] Keep using the same root file: /Users/janosh/Desktop/test.tex.
[10:34:25] Start formatting with latexindent.
[10:34:25] Formatting with command latexindent -c,/Users/janosh/Desktop/,/Users/janosh/Desktop/__latexindent_temp.tex,-y=defaultIndent: '  '
[10:34:25] Formatted /Users/janosh/Desktop/test.tex
[10:34:25] File watcher - file changed: /Users/janosh/Desktop/test.tex
[10:34:25] Parsing a file and its subfiles: /Users/janosh/Desktop/test.tex
[10:34:25] Parse fls file.
[10:34:25] Cannot find fls file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:25] Auto build started detecting the change of a file: /Users/janosh/Desktop/test.tex
[10:34:25] BUILD command invoked.
[10:34:25] Building root file: /Users/janosh/Desktop/test.tex
[10:34:25] Build root file /Users/janosh/Desktop/test.tex
[10:34:25] Recipe step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079,/Users/janosh/Desktop/test
[10:34:25] cwd: /Users/janosh/Desktop
[10:34:25] LaTeX build process spawned. PID: 7080.
[10:34:26] LaTeX log parsed with 2 messages.
[10:34:26] Recipe returns with error: 12/null. PID: 7080. message: Latexmk: This is Latexmk, John Collins, 26 Dec. 2019, version: 4.67.
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -synctex=1 -interaction=nonstopmode -file-line-error -recorder -output-directory="/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079"  "/Users/janosh/Desktop/test.tex"'
------------
Latexmk: Missing input file: 'point-flow.pdf' from line
  'LaTeX Warning: File `point-flow.pdf' not found on input line 7.'
Latexmk: Log file says output to '/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to '/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.
.
[10:34:26] Cleaning auxiliary files and retrying build after toolchain error.
[10:34:26] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.aux
[10:34:26] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fdb_latexmk
[10:34:26] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.log
[10:34:26] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:26] Recipe step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079,/Users/janosh/Desktop/test
[10:34:26] cwd: /Users/janosh/Desktop
[10:34:26] LaTeX build process spawned. PID: 7083.
[10:34:27] LaTeX log parsed with 2 messages.
[10:34:27] Recipe returns with error: 12/null. PID: 7083. message: Latexmk: This is Latexmk, John Collins, 26 Dec. 2019, version: 4.67.
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -synctex=1 -interaction=nonstopmode -file-line-error -recorder -output-directory="/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079"  "/Users/janosh/Desktop/test.tex"'
------------
Latexmk: Missing input file: 'point-flow.pdf' from line
  'LaTeX Warning: File `point-flow.pdf' not found on input line 7.'
Latexmk: Log file says output to '/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to '/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.
.
[10:34:27] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.log
[10:34:27] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:27] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fdb_latexmk
[10:34:27] File cleaned: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.aux
[10:34:35] Start formatting with latexindent.
[10:34:35] Formatting with command latexindent -c,/Users/janosh/Desktop/,/Users/janosh/Desktop/__latexindent_temp.tex,-y=defaultIndent: '  '
[10:34:36] Formatted /Users/janosh/Desktop/test.tex
[10:34:36] File watcher - file changed: /Users/janosh/Desktop/test.tex
[10:34:36] Parsing a file and its subfiles: /Users/janosh/Desktop/test.tex
[10:34:36] Parse fls file.
[10:34:36] Cannot find fls file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:36] Auto build started detecting the change of a file: /Users/janosh/Desktop/test.tex
[10:34:36] BUILD command invoked.
[10:34:36] Building root file: /Users/janosh/Desktop/test.tex
[10:34:36] Build root file /Users/janosh/Desktop/test.tex
[10:34:36] Recipe step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=/var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079,/Users/janosh/Desktop/test
[10:34:36] cwd: /Users/janosh/Desktop
[10:34:36] LaTeX build process spawned. PID: 7088.
[10:34:36] File watcher - file changed: /Users/janosh/Desktop/test.tex
[10:34:36] Parsing a file and its subfiles: /Users/janosh/Desktop/test.tex
[10:34:36] Parse fls file.
[10:34:36] Fls file found: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:36] Parse aux file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.aux
[10:34:36] Auto Build Run is temporarily disabled during a second.
[10:34:37] VIEW command invoked.
[10:34:37] Found root file from active editor: /Users/janosh/Desktop/test.tex
[10:34:37] Keep using the same root file: /Users/janosh/Desktop/test.tex.
[10:34:37] Serving PDF file at http://localhost:54266/viewer.html?file=pdf..JTJGdmFyJTJGZm9sZGVycyUyRjVzJTJGN19sZzdmOXM1Nzcwd2ttcDJyeXNzdzQ0MDAwMGduJTJGVCUyRnRtcC03MDE5LTdwQ0t0SEs1NzA3OSUyRnRlc3QucGRm
[10:34:37] The encoded path is /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf
[10:34:37] Open PDF tab for /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf
[10:34:37] Added to file watcher: /Users/janosh/Desktop/point-flow.pdf
[10:34:37] LaTeX log parsed with 0 messages.
[10:34:37] Recipe of length 1 finished. PID: 7088.
[10:34:37] Successfully built /Users/janosh/Desktop/test.tex.
[10:34:37] No PDF viewer connected for /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf
[10:34:37] Parse fls file.
[10:34:37] Fls file found: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.fls
[10:34:37] Parse aux file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.aux
[10:34:37] Handle data type: open
[10:34:37] Preview PDF file: /var/folders/5s/7_lg7f9s5770wkmp2ryssw440000gn/T/tmp-7019-7pCKtHK57079/test.pdf
[10:34:37] Handle data type: request_params
[10:34:37] Found root file from active editor: /Users/janosh/Desktop/test.tex
[10:34:37] Keep using the same root file: /Users/janosh/Desktop/test.tex.
[10:34:38] Handle data type: loaded
[10:34:40] File watcher - file deleted: /Users/janosh/Desktop/point-flow.pdf
[10:36:21] Found root file from active editor: /Users/janosh/Desktop/test.tex
[10:36:21] Keep using the same root file: /Users/janosh/Desktop/test.tex.
```
</details>

### Developer Tools Console [Required]

<details>

<summary>VS Code Logs</summary>

```
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.yaml.
Old grammar file: file:///Applications/Visual%20Studio%20Code.app/Contents/Resources/app/extensions/yaml/syntaxes/yaml.tmLanguage.json.
New grammar file: file:///Users/janosh/.vscode/extensions/redhat.vscode-yaml-0.10.0/syntaxes/yaml.tmLanguage.json
register @ TMScopeRegistry.ts:45
s @ TMGrammarFactory.ts:73
_getOrCreateGrammarFactory @ abstractTextMateService.ts:246
async function (async)
_getOrCreateGrammarFactory @ abstractTextMateService.ts:235
(anonymous) @ abstractTextMateService.ts:271
_registerDefinitionIfAvailable @ abstractTextMateService.ts:269
(anonymous) @ abstractTextMateService.ts:151
_handle @ extensionsRegistry.ts:142
acceptUsers @ extensionsRegistry.ts:133
_handleExtensionPoint @ abstractExtensionService.ts:397
_doHandleExtensionPoints @ abstractExtensionService.ts:343
_startLocalExtensionHost @ extensionService.ts:586
_scanAndHandleExtensions @ extensionService.ts:556
async function (async)
_scanAndHandleExtensions @ extensionService.ts:501
_initialize @ abstractExtensionService.ts:112
(anonymous) @ extensionService.ts:138
requestIdleCallback (async)
t.runWhenIdle @ async.ts:722
(anonymous) @ extensionService.ts:137
Promise.then (async)
V @ extensionService.ts:135
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:209
_createServiceInstanceWithOwner @ instantiationService.ts:198
_createAndCacheServiceInstance @ instantiationService.ts:187
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:261
(anonymous) @ workbench.ts:139
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:132
open @ desktop.main.ts:124
async function (async)
open @ desktop.main.ts:112
t.main @ desktop.main.ts:329
(anonymous) @ workbench.js:58
processTicksAndRejections @ internal/process/task_queues.js:85
Promise.then (async)
(anonymous) @ workbench.js:54
(anonymous) @ bootstrap-window.js:136
t._invokeFactory @ loader.js:1084
t.complete @ loader.js:1094
s._onModuleComplete @ loader.js:1715
s._onModuleComplete @ loader.js:1727
s._resolve @ loader.js:1675
s.defineModule @ loader.js:1322
r @ loader.js:1765
c @ loader.js:829
(anonymous) @ workbench.common.main.ts:132
(anonymous) @ fake:1
t._createAndEvalScript @ loader.js:832
(anonymous) @ loader.js:814
u @ loader.js:933
(anonymous) @ loader.js:949
readFileAfterClose @ internal/fs/read_file_context.js:61
console.ts:137 [Extension Host] (node:7390) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
_doInvokeHandler @ rpcProtocol.ts:409
_invokeHandler @ rpcProtocol.ts:394
_receiveRequest @ rpcProtocol.ts:310
_receiveOneMessage @ rpcProtocol.ts:237
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:463
_receiveMessage @ ipc.net.ts:747
(anonymous) @ ipc.net.ts:602
fire @ event.ts:587
acceptChunk @ ipc.net.ts:240
(anonymous) @ ipc.net.ts:201
t @ ipc.net.ts:29
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166
console.ts:137 [Extension Host] (node:7390) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
_doInvokeHandler @ rpcProtocol.ts:409
_invokeHandler @ rpcProtocol.ts:394
_receiveRequest @ rpcProtocol.ts:310
_receiveOneMessage @ rpcProtocol.ts:237
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:463
_receiveMessage @ ipc.net.ts:747
(anonymous) @ ipc.net.ts:602
fire @ event.ts:587
acceptChunk @ ipc.net.ts:240
(anonymous) @ ipc.net.ts:201
t @ ipc.net.ts:29
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166
mainThreadExtensionService.ts:65 [[object Object]]String was not valid TexCount output
$onExtensionRuntimeError @ mainThreadExtensionService.ts:65
_doInvokeHandler @ rpcProtocol.ts:409
_invokeHandler @ rpcProtocol.ts:394
_receiveRequest @ rpcProtocol.ts:310
_receiveOneMessage @ rpcProtocol.ts:237
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:463
_receiveMessage @ ipc.net.ts:747
(anonymous) @ ipc.net.ts:602
fire @ event.ts:587
acceptChunk @ ipc.net.ts:240
(anonymous) @ ipc.net.ts:201
t @ ipc.net.ts:29
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166
mainThreadExtensionService.ts:66 Error: String was not valid TexCount output
	at WordCounter.parseTexCount (/Users/janosh/.vscode/extensions/tecosaur.latex-utilities-0.3.7/out/src/components/wordCounter.js:120)
	at ChildProcess.<anonymous> (/Users/janosh/.vscode/extensions/tecosaur.latex-utilities-0.3.7/out/src/components/wordCounter.js:67)
	at ChildProcess.emit (events.js:203)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:272)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:66
_doInvokeHandler @ rpcProtocol.ts:409
_invokeHandler @ rpcProtocol.ts:394
_receiveRequest @ rpcProtocol.ts:310
_receiveOneMessage @ rpcProtocol.ts:237
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:463
_receiveMessage @ ipc.net.ts:747
(anonymous) @ ipc.net.ts:602
fire @ event.ts:587
acceptChunk @ ipc.net.ts:240
(anonymous) @ ipc.net.ts:201
t @ ipc.net.ts:29
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166
webviewElement.ts:162 [Embedded Page] PDF 2692e69fa556234a134a178de7413530 [1.5 pdfTeX-1.40.21 / TeX] (PDF.js: 2.3.200)
console.ts:137 [Extension Host] Sync : Connected with user : 'janosh'
```
</details>



## Screenshots

Shows preview despite preview being disabled.

![preview-despite-disabled](https://user-images.githubusercontent.com/30958850/91152504-52c4e980-e6bf-11ea-90a2-78f4f997788a.png)




## Desktop [Required]
 - OS: macOS 10.15.6
 - VS Code version: 1.46.1
 - Extension version: 8.11.1
 - TeX distribution version: TeX Live 2019

## Additional questions

### Are you using VSCodium?

**No**

### Are you using LaTeX Workshop with VS Code Remote?

**No**

--- a/package.json
+++ b/package.json
@@ -1557,6 +1557,11 @@
           "default": true,
           "markdownDescription": "Enable preview for `\\includegraphics` completion."
         },
+        "latex-workshop.intellisense.includegraphics.preview.pdf.enabled": {
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Enable preview for `\\includegraphics` of PDF file."
+        },
         "latex-workshop.message.badbox.show": {
           "type": "boolean",
           "default": true,

--- a/src/providers/preview/graphicspreview.ts
+++ b/src/providers/preview/graphicspreview.ts
@@ -69,6 +69,11 @@ export class GraphicsPreview {
             return undefined
         }
         if (/\.pdf$/i.exec(filePath)) {
+            const configuration = vscode.workspace.getConfiguration('latex-workshop')
+            const preview = configuration.get('intellisense.includegraphics.preview.pdf.enabled') as boolean
+            if (!preview) {
+                return undefined
+            }
             const cacheKey = JSON.stringify([filePath, pageNumber])
             const cache = this.pdfFileCacheMap.get(cacheKey)
             const cacheFileName = cache?.cacheFileName ?? this.newSvgCacheName()

